csi.proto

```diff
@@ -271,15 +271,18 @@ message VolumeMetadata {
 }
 
 // A standard way to encode credential data. The total bytes of the values in
-// the Data field must be less than MaxSecretSize bytes.
+// the Data field must be less than 1 Mebibyte.
 message Credentials {
-  // Data contains the credential data, for example username and password. Each
-  // key must consist of alphanumeric characters, '-', '_' or '.'. The
-  // serialized form of the secret data is a base64 encoded string, representing
-  // the arbitrary (possibly non-string) data value here.
-  // This information is sensitive and should be treated as such (not logged,
-  // etc.)
-  map<string, bytes> Data = 1;
+  // Data contains the credential data, for example username and password.
+  // Each key must consist of alphanumeric characters, '-', '_' or '.'.
+  // Each value MUST contain a valid string. An SP MAY choose to accept binary
+  // (non-string) data by using a binary-to-text encoding scheme, like base64.
+  // An SP SHALL advertise the requirements for credentials in documentation.
+  // COs SHALL permit users to pass through the required credentials.
+  // This information is sensitive and MUST be treated as such (not logged,
+  // etc.) by the CO.
+  // This field is REQUIRED.
+  map<string, string> Data = 1;
```

Comment:
`map<string, bytes>` was what I proposed initially but I realized that makes the normal use case harder.

> Using map<string, string> here will force all COs to do base64 encoding to convert a binary secret to a string

To be very clear, `map<string, string>` requires no extra work (base64 encoding) when passing string values. Only if the value contains non-text (aka non-string, binary) data will data need to be base64 encoded.

`map<string, bytes>`, on the other hand, makes passing string data more complicated: to pass a string via a byte array you must always specify the character-encoding (US-ASCII, UTF-8, etc.). Either the this character-encoding would be published as part of the documentation, and clients would have to handle it, or we'd have to create another field to allow it to be set dynamically. Either way it would necessarily make the passing string data more complicated, whereas the alternative, `map<string, string>`, makes only binary data needs to be special cased (documentation indicates the value for some key is binary and must be base64 encoded).

Because the majority of cases will likely want to pass in a simple username/password string, I strongly prefer optimizing for that use case, rather than the binary blob use case.